red/runtime/nodes/Node.js

```diff
@@ -23,11 +23,14 @@ var Log = require("../log");
 var context = require("./context");
 var flows = require("./flows");
 
+var _timeout = undefined;
+
 function Node(n) {
     this.id = n.id;
     this.type = n.type;
     this.z = n.z;
     this._closeCallbacks = [];
+    this._msg_in = undefined;
```

Comment:
A node could handle more than one message at a time. We cannot have a node-level property storing just the most recent message. I can't see in the code why we have to store the full message outside of the on-input message handler.

Reply:
The stored message is used for logging correlation of the last input message & output messages as described  in the following description of Design Note in `Node.prototype.metric` when old-style `Node.prototype.send` is used.

> (HN): According to the discussion held on May 17th, the correlation information among an 
> input message and output messages are expected to be used for logging purpose, and a 
> output message correlate with the latest input message received.

Regarding use of node-level property, we can use closure making the propety enclosing scope variable of `Node.prototype.on` and `Node.prototype.send` insted.